connectors/sources/google_drive.py

```diff
@@ -1100,9 +1103,16 @@ async def prepare_file(self, file, paths):
             # Read more: https://developers.google.com/drive/api/guides/shared-drives-diffs
             permissions = file.get("permissions", [])
             if not permissions:
-                permissions = await self._get_permissions_on_shared_drive(
-                    file_id=file_id
-                )
+                try:
+                    permissions = await self._get_permissions_on_shared_drive(
+                        file_id=file_id
+                    )
+                except HTTPError as exception:
```

Comment:
not blocking: I wonder if some of the HTTPError exceptions can be fixed by retrying the request or reauthenticate it?

Reply:
> I wonder if some of the HTTPError exceptions can be fixed by retrying the request
Nope, we already have a re-try logic at a lower level. This `try/except` method is only supposed to gracefully handle failure scenarios after all retries are exhausted. The underlying issue is with scenario where user doesn't grant sufficient permissions to a file, enable dls, and so permissions fetching will always fail. 

> if you handle only a specific case like 403 error then it's worth adding an extra `try:` where you log that specific error.
Yes, exactly I only handle 403s in this clause. I can't really add extra logging at the route of the exception as this is handled by a library. Not sure if I understand what you mean here, can you elaborate. For context the logged `exception` will print the status code, and a detailed error message.

Reply:
> I can't really add extra logging at the root of the exception as this is handled by the aiogoogle library. Not sure if I understand what you mean here, can you elaborate.

what I mean is: 
```python
   try:
    .....
   except HTTPError as e:
     if e.reg.status == 403:
       self._logger.warning("Unable to fetch permissions due to 403 error. Exception {exception}")
    else:
      self._logger.error("Unable to fetch permissions due exception {exception}")
```

In this case, we can use different log levels or even handle those errors differently if we need. 
"Need" - is the main question here :)

Reply:
That's a good point, we might want to make the non-403s more visible to the user. I updated the PR! :)